### ISSUE TITLE
[BUG]Casting string column with all nulls to datetime fails.

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#5939]
**Describe the bug**
Casting string column with all nulls to DateTime fails.

**Steps/Code to reproduce bug**
```python
>>> cudf.Series([None,None],dtype='str').astype('datetime64[ns]')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/series.py", line 2141, in astype
    raise e
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/series.py", line 2133, in astype
    data = self._column.astype(dtype)
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/column/column.py", line 884, in astype
    return self.as_datetime_column(dtype, **kwargs)
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/column/string.py", line 4551, in as_datetime_column
    return self.as_numerical_column(dtype, **kwargs)
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/column/string.py", line 4522, in as_numerical_column
    fmt = datetime.infer_format(self[self.notna()][0])
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/column/column.py", line 477, in __getitem__
    return self.element_indexing(arg)
  File "/raid/vjawa/conda/envs/cudf_015_nightly_aug_11/lib/python3.7/site-packages/cudf/core/column/column.py", line 468, in element_indexing
    raise IndexError("single positional indexer is out-of-bounds")
IndexError: single positional indexer is out-of-bounds
```
**Expected behavior**
Like for non empty columns or non string columns. 
```python
>>> import cudf
>>> cudf.Series([None,'2015-02-04'],dtype='str').astype('datetime64[ns]')
0                             <NA>
1    2015-02-04 00:00:00.000000000
dtype: datetime64[ns]
>>> cudf.Series([None,None],dtype='str').astype('datetime64[ns]')
```

```python
>>> cudf.Series([None,None],dtype='int').astype('datetime64[ns]')
0    <NA>
1    <NA>
dtype: datetime64[ns]
```
 - Method of cuDF install: [conda]
```python
cudf                      0.15.0a200811   py37_gef1629c17_4253    rapidsai-nightly
libcudf                   0.15.0a200811   cuda10.2_gef1629c17_4253    rapidsai-nightly
```

**Comment by galipremsagar** [comment]
I think I have a fix for this issue, can we do this for 0.15 @kkraus14 ?

**Comment by kkraus14** [comment]
Yes
